### PR TITLE
Feature/wellbore mesh, Traction BC, convert {tensor input}

### PR DESCRIPTION
**Pull request by rrsettgast**
Replacement pull request for #1338 with cleaner history appropriate for non-squash merge commit.

- [x] Implements internal mesh generation for cylinder/wellbore
- [x] integrated tests for wellbore
- [x] Changes input for tensor types to conform with array input notation using `{}` 
- [x] Traction boundary condition type to encapsulate the BC application.

### CODE REVIEW

File: src/coreComponents/physicsSolvers/solidMechanics/integratedTests/OpenWellbore.xml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+      name="mesh1"
+      elementTypes="{ C3D8 }"
+      radius="{ 0.1, 2 }"
+      theta="{ 0, 180 }"
+      zCoords="{ 0, 3 }"
+      nr="{ 5 }"
+      nt="{ 8 }"
+      nz="{ 3 }"
+      trajectory="{ { -0.1, 0.0, 0.0 },
+                    {  0.1, 0.0, 1.0 } }"
```
Comment (sytuannguyen) [inline src/coreComponents/physicsSolvers/solidMechanics/integratedTests/OpenWellbore.xml:39]:
```suggestion
                    {  0.1, 0.0, 3.0 } }"
```

File: src/coreComponents/physicsSolvers/solidMechanics/integratedTests/casedWellbore.xml [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+      name="mesh1"
+      elementTypes="{ C3D8, C3D8, C3D8 }"
+      radius="{ 0.1, 0.11, 0.15, 2.0 }"
+      theta="{ 0, 360 }"
+      zCoords="{ 0, 4 }"
+      nr="{ 2, 2, 5 }"
+      nt="{ 16 }"
+      nz="{ 4 }"
+      trajectory="{ { 0.0, 0.0, 0.0 },
+                    { 0.0, 0.0, 1.0 } }"
```
Comment (sytuannguyen) [inline src/coreComponents/physicsSolvers/solidMechanics/integratedTests/casedWellbore.xml:39]:
```suggestion
                    { 0.0, 0.0, 4.0 } }"
```

**Review by sytuannguyen — APPROVED** [review]

File: src/coreComponents/fieldSpecification/FieldSpecificationBase.hpp [hunk 3]
```diff
@@ -446,8 +461,10 @@ class FieldSpecificationBase : public dataRepository::Group
   }
 
   /**
-   * Accessor
-   * @return const m_component
+   * Accessing the considered component.
+   * The 0, 1 and 2 values are for the traction in the three orthogonal directions in the global cartesian coordinate system.
+   * The -1 value is for the normal traction.
```
Comment (CusiniM) [inline src/coreComponents/fieldSpecification/FieldSpecificationBase.hpp:466]: How s the -1 value used?

**Review by CusiniM — APPROVED** [review]
Just some minor comments. We should do what you did with the traction b.c. also with the others.

File: src/coreComponents/fieldSpecification/FieldSpecificationBase.hpp (hunk 3 above)
Comment (rrsettgast) [reply src/coreComponents/fieldSpecification/FieldSpecificationBase.hpp:466]: no longer used....removed.

File: src/coreComponents/dataRepository/Wrapper.hpp [hunk 4]
```diff
@@ -713,7 +714,10 @@ class Wrapper final : public WrapperBase
       }
       else
       {
-        xmlWrapper::readAttributeAsType( reference(), getName(), targetNode, getDefaultValueStruct() );
+        m_successfulReadFromInput = xmlWrapper::readAttributeAsType( reference(),
+                                                                     getName(),
+                                                                     targetNode,
+                                                                     getDefaultValueStruct() );
```
[the remark this answers is not in the document]
Comment (rrsettgast) [reply src/coreComponents/dataRepository/Wrapper.hpp:720]: You can if it is desired. This is really for optional input, that have other input that they are dependent on. So if you don't want to have one of the input if the other was read....or you want to check if one input was read if another was read....etc. Really just a tool for better error checking in the `postProcessInput()` functions.

File: src/coreComponents/fieldSpecification/TractionBoundaryCondition.cpp [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+
+//  localIndex const numStressFunctionsNames = m_stressFunctionNames.size();
+//  GEOSX_ERROR_IF( numStressFunctionsNames > 0 && numStressFunctionsNames<6,
+//                  "Either 0 or 6 stress functions must be specified using stressFunctions" );
+//
+//  if( numStressFunctionsNames==6 )
+//  {
+//    m_useStressFunctions = true;
+//  }
```
[the remark this answers is not in the document]
Comment (rrsettgast) [reply src/coreComponents/fieldSpecification/TractionBoundaryCondition.cpp:95]: I am actually leaving this here intentionally. We have to enable reading the stress input from a space table....which I started to do here, but it would be better for a follow on PR.

File: src/coreComponents/fieldSpecification/TractionBoundaryCondition.cpp [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+{
+//  FunctionManager const & functionManager = getGlobalState().getFunctionManager();
+//
+//  if( m_useStressFunctions )
+//  {
+//    for( localIndex i=0; i<6; ++i )
+//    {
+//      m_stressFunctions[i] = &( functionManager.getGroup< TableFunction >( m_stressFunctionNames[i] ) );
+//    }
+//  }
```
[the remark this answers is not in the document]
Comment (rrsettgast) [reply src/coreComponents/fieldSpecification/TractionBoundaryCondition.cpp:108]: same as above.

File: src/coreComponents/mesh/generators/InternalMeshGenerator.cpp [hunk 7]
```diff
@@ -701,36 +614,34 @@ void InternalMeshGenerator::generateMesh( DomainPartition & domain )
...
+
+                // Fix local connectivity for single theta (y) partition (radial meshes only)
+
+//                setConnectivityForPeriodicBoundaries( i, j, k,
+//                                                      iblock, jblock, kblock,
+//                                                      globalIJK,
+//                                                      numElemsInDirForBlock,
+//                                                      numNodesInDir,
+//                                                      firstElemIndexInPartition,
+//                                                      nodeOfBox );
```
[the remark this answers is not in the document]
Comment (rrsettgast) [reply src/coreComponents/mesh/generators/InternalMeshGenerator.cpp:627]: yes...this was actually incorrectly commented out.
